token/js/src/extensions/immutableOwner.ts

```diff
@@ -0,0 +1,24 @@
+import { struct } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import { PublicKey } from '@solana/web3.js';
+import { Mint } from '../state/mint';
+import { ExtensionType, getExtensionData } from './extensionType';
+
+/** MintCloseAuthority as stored by the program */
+export interface ImmutableOwner {
+    owner: PublicKey;
+}
+
+/** Buffer layout for de/serializing a mint */
+export const ImmutableOwnerLayout = struct<ImmutableOwner>([publicKey('owner')]);
+
+export const IMMUTABLE_OWNER_SIZE = ImmutableOwnerLayout.span;
+
+export function getImmutableOwner(mint: Mint): ImmutableOwner | null {
```

Comment:
The `ImmutableOwner` extension only works on an account, and not a mint: https://github.com/solana-labs/solana-program-library/blob/24baf875e9e19c26d694d28c557d33848c3a9180/token/program-2022/src/extension/mod.rs#L695 -- and essentially all we might need to know is if the account's owner is immutable or not